### PR TITLE
check for muxed support in collectibles

### PR DESCRIPTION
**Pull request by piyalbasu**
Related to #2513 

Examples:

Screenshot 2026-01-20 at 9 03 24 PM
Screenshot 2026-01-20 at 9 02 57 PM

This pull request enhances the handling and testing of sending collectibles (NFTs) and tokens in the extension, especially regarding contract support for muxed (M-) addresses and memo field behavior. It introduces a new helper function to consistently determine the correct contract ID for both tokens and collectibles, and updates the UI logic and tests to account for various combinations of asset type, contract capabilities, and recipient address type.

Key changes include:

**Feature Enhancements and Bug Fixes:**

- Improved collectible send flow to correctly handle cases where the contract does or does not support muxed addresses, including disabling the send action and memo fields as appropriate, and updating UI warnings and validation logic. [[1]](diffhunk://#diff-882ef6fa343af3799ac56d363056e440f59ca5226296a0e40666324be0255c4dR182-R454) [[2]](diffhunk://#diff-c515baeaf91c6b93f1445a961fcd187596274cca28e6543483b28002ac995061L197-R221) [[3]](diffhunk://#diff-c515baeaf91c6b93f1445a961fcd187596274cca28e6543483b28002ac995061L438-R461)
- Updated memo field logic so that it is only enabled when supported by both the asset type and contract capabilities, with new UI states and error messages for unsupported scenarios. [[1]](diffhunk://#diff-882ef6fa343af3799ac56d363056e440f59ca5226296a0e40666324be0255c4dR182-R454) [[2]](diffhunk://#diff-c515baeaf91c6b93f1445a961fcd187596274cca28e6543483b28002ac995061L197-R221)

**Refactoring and Code Reuse:**

- Introduced a new helper function, `getContractIdFromTransactionData`, to centralize logic for extracting the contract ID from transaction data for both tokens and collectibles, replacing previous ad-hoc implementations. [[1]](diffhunk://#diff-c515baeaf91c6b93f1445a961fcd187596274cca28e6543483b28002ac995061L21-R25) [[2]](diffhunk://#diff-84697be805778c1ccca0e2bf770a22c8f5455dc7010b847c0bb95ce7be40c3e9L19-R19) [[3]](diffhunk://#diff-8b2adf3e68839a009c6962b570d8130bff73730a3cb5a14ee77e5df27f5430edR16)
- Refactored affected components (`SendAmount`, `ReviewTransaction`) to use the new helper, simplifying and unifying contract ID extraction logic. [[1]](diffhunk://#diff-c515baeaf91c6b93f1445a961fcd187596274cca28e6543483b28002ac995061L135-R155) [[2]](diffhunk://#diff-84697be805778c1ccca0e2bf770a22c8f5455dc7010b847c0bb95ce7be40c3e9L112-R128)

**Test Coverage:**

- Added comprehensive tests for `getContractIdFromTransactionData`, covering all combinations of collectible and token flags, asset formats, and edge cases.
- Expanded E2E tests for sending collectibles to cover all scenarios of muxed address support, memo field enablement, and disabled send actions, ensuring robust validation of the new logic.

**UI/UX Improvements:**

- Updated the continue/review send button to be disabled when attempting to send a collectible to an M-address if the contract does not support muxed addresses, preventing invalid transactions.
- Improved warning banners and error messages in the UI to clearly communicate unsupported operations to the user.

These changes ensure that the extension handles all combinations of tokens, collectibles, muxed address support, and memo field logic in a consistent, user-friendly, and robust manner.

### CODE REVIEW

File: extension/e2e-tests/sendCollectible.test.ts [hunk 1]
```diff
@@ -177,3 +179,276 @@ test("Send collectible without metadata", async ({
...
+  await expect(page.getByTestId("SelectedCollectible")).toBeVisible();
+
+  // Send to M address (muxed address)
+  await page.getByTestId("address-tile").click();
+  await page.getByTestId("send-to-input").fill(TEST_M_ADDRESS);
+  await page.getByText("Continue").click({ force: true });
+
+  // Wait for contract check to complete
+  // Since contract supports muxed (true), there should be NO warning banner
+  await page.waitForTimeout(2000);
```
Comment (aristidesstaffieri) [inline extension/e2e-tests/sendCollectible.test.ts]: is this needed? Imo this typically implies that there is something happening without UI representation because in theory you should be able to wait for the expected dom element instead(action -> loading state -> final state).

File: extension/e2e-tests/sendCollectible.test.ts [hunk 2]
```diff
@@ -177,3 +179,276 @@ test("Send collectible without metadata", async ({
...
+  ).toBeVisible({ timeout: 15000 });
+  await page.getByText("Cancel").click();
+
+  // Click Review Send
+  const reviewSendButton = page.getByTestId("send-collectible-btn-continue");
+  await expect(reviewSendButton).toBeEnabled({ timeout: 10000 });
+  await reviewSendButton.click({ force: true });
+
+  // Wait for review sheet to open
+  await expect(page.getByText("You are sending")).toBeVisible({
```
Comment (aristidesstaffieri) [inline extension/e2e-tests/sendCollectible.test.ts]: this timeout is long and does not really seem to be tied to any state, is it possible to await the loading state instead?

Comment (piyalbasu) [reply extension/e2e-tests/sendCollectible.test.ts]: This was a bit of a hack picked up from another test to compensate for network latency. Removing this here: https://github.com/stellar/freighter/pull/2519/commits/9bdc9722e355734d3505b61163c8dbf66cb3211e

When I merge my e2e tests refresh, these hacks will no longer be necessary 🎉
